lib/taskana-core/src/main/java/pro/taskana/task/internal/ServiceLevelHandler.java

```diff
@@ -159,17 +159,53 @@ private TaskImpl updatePlannedDueOnTaskUpdate(
         Instant planned =
             (converter.subtractWorkingDaysFromInstant(
                 newTaskImpl.getDue(), durationPrioHolder.getDuration()));
-        if (newTaskImpl.getPlanned() != null && !planned.equals(newTaskImpl.getPlanned())) {
+        // planned may not be the same as newTaskImpl.planned and the request may nevertheless be
```

Comment:
For the sake of readability: 
- Move this important comment as a JavaDoc comment for the method `plannedDateViolatesServiceLevel`
- include `newTaskImpl.getPlanned() != null && !planned.equals(newTaskImpl.getPlanned()` into the method `plannedDateViolatesServiceLevel` (for completeness and reusability)
- Maybe make the method `plannedDateViolatesServiceLevel` a void return type and let it throw the exception? - I am not sure about this last one


Would this help? What do you think?

Reply:
done